src/js/components/DataTable/buildState.js

```diff
@@ -173,33 +173,45 @@ export const buildFooterValues = (columns, data) => {
 
 // looks at the groupBy property of each data object and returns an
 // array with one item for each unique value of that property.
-export const buildGroups = (columns, data, groupBy) => {
+export const buildGroups = (columns, data, groupBy, primaryProperty) => {
   let result;
-  if (groupBy) {
+  if (groupBy?.property || typeof groupBy === 'string') {
     result = [];
     const groupMap = {};
     data.forEach((datum) => {
+      const key = datumValue(datum, primaryProperty);
+      const isGroup = (key && groupBy.expandable?.includes(key));
+
       const groupByProperty = groupBy.property ? groupBy.property : groupBy;
-      const groupValue = datumValue(datum, groupByProperty);
-      if (!groupMap[groupValue]) {
-        const group = { data: [], datum: {}, key: groupValue };
+      const groupValue = isGroup ? key : datumValue(datum, groupByProperty);
+      if (!groupMap[isGroup ? key : groupValue]) {
```

Comment:
I think `groupValue` already is `key` when `isGroup`.
```suggestion
      if (!groupMap[groupValue]) {
```

Reply:
very true. Thanks